serverless/question-service/functions/utils/concurrency.ts

```diff
@@ -0,0 +1,17 @@
+import { chunk } from "lodash";
+
+export async function processConcurrently(
+  funcs: (() => Promise<any>)[],
+  chunkSize = 20
+) {
+  const safeChunkSize = Math.min(chunkSize, 20);
+  const chunks = chunk(funcs, safeChunkSize);
+  for (const chunk of chunks) {
+    const promises = chunk.map((func) => func());
+    await Promise.allSettled([...promises, wait(500)]);
+  }
+}
+
```

Comment:
does this function process the async `funcs` at `chunkSize` concurrent requests at a time??? q cool sia where u find this function one